runners/genlayer-py-std/src-emb/genlayer_embeddings/_nn/builder.py

```
@@ -37,24 +78,61 @@ def add_decl(self, expr) -> str:
 			d = '\tpass\n'
 		else:
-			d = '\n'.join('\t' + line for line in d.split('\n') if line.strip())
-		d = f'def main({params_str}):\n{d}'
+			d = '\n'.join('\t' + line for line in d.split('\n') if line.strip()) + '\n'
 
+		self._prelude.append(f'def main({params_str}):\n')
+		d = ''.join(self._prelude) + d
+
+		return d
+
+	def finish(self, parameters: list[str] = []) -> typing.Callable:
```

Ruff (0.12.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)